### DISCUSSION TITLE
Add locals-usa.m3u

### DISCUSSION THREAD

**Discussion by MapGuy11** [iptv-org/community#204] [Ideas]
I plan to drop a motherload of locals and wonder if you want to move all locals to a new .m3u list like locals-USA.m3u instead of everything in the USA.m3u file. Let me know your thoughts.

**Comment by BellezaEmporium** [comment]
Keep those in the us.m3u file. I know this file is a big pile but I don't think we're planning to make a difference.

**Reply by MapGuy11** [reply]
Ok. Some big pull requests will be coming in the coming days as I have some more time to drop more.

**Reply by MapGuy11** [reply]
Can I just add them all to the start of the file and the bot will re-sort them correct?

**Reply by BellezaEmporium** [reply]
Better put them at the end, like what the majority of us do. The bot re-sorts them, that's correct.

**Reply by MapGuy11** [reply]
ok

**Comment by themak1985** [comment]
They should all go in us.m3u, but they should be marked as Local for group-title.  I would like to standardize how we name US Locals. I suggest the following options and examples:

`ABC New York City (WABC)`
`[Channel] [City] [(Call Sign)]`

If we must include channel number, I would say

`[Channel] [City] [(Call Sign ChannelNumber)]`
`NBC Lx New York City (WNBC-DT3 4.3)`

Adding a -
`[Channel] [City] [(Call Sign ChannelNumber)]`
`NBC Lx - New York City (WNBC-DT3 4.3)`

**Reply by Dum4G** [reply]
Imagine if it's also geoblocked and not 24/7. I guess local channels have cities in the name itself or at least you know what you search for

**Reply by BellezaEmporium** [reply]
> 
> 
> Imagine if it's also geoblocked and not 24/7. I guess local channels have cities in the name itself or at least you know what you search for

By it's call sign, you can actually know from where it's transmitted, and which channels are into this TV station.

Here, @themak1985 simply tells us how he wants the local US stations to be named (you can also put that to an certain extent with Mexican channels being broadcasted from the US (X--- - TDT, see http://www.tdt.mx/index.php)

Of course, if we find out that those channels aren't geoblocked and not 24/7, we can flag them as such.

As far as I see @MapGuy11 is from the US, so he's pretty much a valuable element to us for knowing if the channel works inside the US territory, and then, we can see if it works outside of the US territory.

**Reply by MapGuy11** [reply]
Yes I am in the US

**Reply by themak1985** [reply]
@MapGuy11 what do you think of the naming scheme so that all the channels will be listed near each other? Alternatively - I have thought about listing the city first.

**Reply by MapGuy11** [reply]
@themak1985 I don't have a problem with the naming scheme I think it will work great and keep everything organized!

**Comment by freearhey** [comment]
@MapGuy11 @themak1985 Do you think it will be okay if we swap Call Sign and Network/City?

`[Call Sign] ([Network] [City])`
`WABC-TV (ABC New York City)`

This way to generate `tvg-id`/`tvg-name` it would be enough to erase everything in brackets:

WABC-TV ~(ABC New York City)~ => `tvg-name="WABC-TV"` => `tvg-id="WABCTV.us"`

This should allow us to automate the entire process and add a program for more channels.

**Reply by themak1985** [reply]
Adding the call sign might make it more difficult for users to find their channel. Instinct is to go to ABC first, then find your local ABC,  I don't even know all my call signs for local channels here. However, if it will make generating the programs easier, I say go ahead. People can just google what their local call signs are.

**Reply by MapGuy11** [reply]
TVG-ID should be WABCDT then DT 2 is WABCDT2

**Reply by themak1985** [reply]
Thats right - call signs and TVG-ID are usually different.

**Reply by freearhey** [reply]
@MapGuy11 I took this call sign from lyngsat: https://www.lyngsat.com/tvchannels/us/WABC-TV.html 

In the program guide only WABC-TV or WABC is used for this channel as well: https://tvtv.us/ny/new-york/10001/lu6350

**Reply by MapGuy11** [reply]
Ok it's just the Digital TV Standard is DT1, DT2, DT3 and so on.

**Reply by freearhey** [reply]
We can use any `tvg-id`, I just decided to use lyngsat as a source when creating `iptv-org/epg`, since it has information about most channels.

If you know any more accurate source, let me know.

**Reply by MapGuy11** [reply]
This source is great for XMLTV Guide: https://tvlistings.zap2it.com/
I am not sure how your epg scrapper works ,but I can help getting this one integrated

**Reply by freearhey** [reply]
@MapGuy11 Unfortunately, this site does not work without a VPN.
![image](https://user-images.githubusercontent.com/7253922/132423248-031bb1b9-4a25-41ce-b908-b90a59075de8.png)

And the problem is not to find the source of the program, [iptv-org/epg](https://github.com/iptv-org/epg) already contains a daily program for thousands of channels: https://iptv-org.github.io/epg/index.html. 

The only difficulty is to find a single source with all the detailed information about each channel and its correct name.

**Reply by what-a** [reply]
I'm thinking we should drop most all instances of "-TV" or "-DT1" from the call signs. It really doesn't add any value and it takes up room. I'm fine with leaving "-DT2", "-DT3", etc.  But there is no need for "-TV" or "-DT1" in my opinion (and I'm in the USA).  I can perform the clean up, but wanted to hear if there are any objections.

**Comment by jkemp814** [comment]
In most cases, that is the call sign. (with -TV or DT1) Also, the EPG would have to be changed.

**Reply by what-a** [reply]
Fair enough.  I'll stand down.
